pkg/controller/petset/iterator.go

```diff
@@ -0,0 +1,168 @@
+package petset
+
+import (
+	"fmt"
+	"sort"
+
+	"github.com/golang/glog"
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/apis/apps"
+	"k8s.io/kubernetes/pkg/controller"
+)
+
+// petGenerator generates pets.
+type petGenerator struct {
+	// ps is the petset for which this generator generates pets
+	ps *apps.PetSet
+	// idMappers are a list of identityMapper interfaces capable of giving
+	// the generates pets an identity
+	idMappers []identityMapper
+}
+
+// newPet generates a new pet using the id string as a unique qualifier
+func (pg *petGenerator) newPet(id string) (*pcb, error) {
```

Comment:
yeah fair enough. in python you give a generator to an iterator, it reminded me of that pattern. its now just a func. I can always expand if it needs to track cross-pet state.